sktime/forecasting/statsforecast.py

```diff
@@ -552,3 +554,124 @@ def get_test_params(cls, parameter_set="default"):
+    >>> y_pred = fitted_model.predict(fh=[1,2,3]) # doctest: +SKIP
+    """
+
+    _tags = {
+        "ignores-exogeneous-X": True,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": True,
+        "python_dependencies": ["statsmodels", "statsforecast"],
+    }
+
+    def __init__(
+        self,
+        season_length: Union[int, List[int]],
+        trend_forecaster=None,
+    ):
+        self.season_length = season_length
+        if trend_forecaster:
+            self.trend_forecaster = trend_forecaster
+            if trend_forecaster.get_tag("scitype:y") == "multivariate":
+                self.set_tags(**{"scitype:y": "multivariate"})
+        else:
+            self.trend_forecaster = StatsForecastAutoETS(model="ZZN")
+
+        super().__init__()
+
+    def _instantiate_model(self):
+        """Create underlying forecaster instance."""
+        from statsforecast.models import _TS, MSTL
+
+        # checks if trend_forecaster is already wrapped with
+        # StatsForecastBackAdapterMSTL
+        if isinstance(self.trend_forecaster, StatsForecastBackAdapter):
+            return MSTL(
+                season_length=self.season_length,
+                trend_forecaster=self.trend_forecaster,
+            )
+        # if trend_forecaster is sktime forecaster
+        elif isinstance(self.trend_forecaster, BaseForecaster):
+            self.trend_forecaster = StatsForecastBackAdapter(self.trend_forecaster)
+        # if trend_forecaster is not StatsForecast forecaster
+        elif not isinstance(self.trend_forecaster, _TS):
+            raise Exception(
+                "The provided forecaster is not compatible with MSTL. Please ensure "
+                "that the forecaster you pass into the model is a sktime forecaster or "
+                "statsforecast forecaster."
+            )
+
+        return MSTL(
+            season_length=self.season_length,
+            trend_forecaster=self.trend_forecaster,
+        )
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for forecasters.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        from sktime.forecasting.theta import ThetaForecaster
```

Comment:
FYI, this is what raises the current exception, as `ThetaForecaster` and `VAR` have a `statsmodels` dependency.

You can work with `_check_soft_dependencies` to have them as part of the test param set only if `statsmodels` is available, or replace with an estimator that doesn't have the `statsmodels` dependency, e.g., `NaiveForecaster`.